### ISSUE TITLE
Why do getCurrentTime(), getDuration(), etc, return a Promise?

### ISSUE THREAD

**Issue by ffxsam** [gajus/youtube-player#49]
Is there any particular reason these methods return a `Promise`? According to the YouTube documentation, they should return numbers.

https://developers.google.com/youtube/iframe_api_reference#Retrieving_video_information

**Comment by resistdesign** [comment]
@ffxsam All of the methods return a Promise because they wait for the Player instance to be ready :)

**Comment by ffxsam** [comment]
But if the player loads up and the video is displayed, by the time the user hits the play button, the player would definitely be ready. Unless I'm misunderstanding something.

**Comment by resistdesign** [comment]
It's not just for the user to control the player, you may also want to do something programmatically as soon as the player is ready.

Also, it loads the YouTube iframe API (which is something that needs to be done at runtime) for you and waits for that as well!

**Comment by ffxsam** [comment]
It's pretty unwieldy to work with, though, when you just want to fetch the duration and current time:

```js
              this.videoPlayer.player.getDuration().then(value => {
                duration = value;
              }).then(() => {
                this.videoPlayer.player.getCurrentTime().then(value => {
                  seconds = value;
                }).then(() => {
                  this.trackVideoAction('pause', {
                    duration,
                    seconds,
                  })
                });
              });
```

**Comment by gajus** [comment]
> It's pretty unwieldy to work with, though, when you just want to fetch the duration and current time:

No reason not to use [async/ await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function).

```js
const duration = await this.videoPlayer.player.getDuration();
const seconds = await this.videoPlayer.player.getCurrentTime();

this.trackVideoAction('pause', {
  duration,
  seconds,
});

```

**Comment by ffxsam** [comment]
Ah! Something I've barely scratched the surface with so far. Thanks for the great suggestion!

**Comment by arckdash** [comment]
Hi, I just starting to use this library, I don't why but the promises always return undefined, any thoughts ?

I quite new about promise, but for some reason I'm getting error when I'm trying to use:

const duration = await this.videoPlayer.player.getDuration();
const seconds = await this.videoPlayer.player.getCurrentTime();

Also I tried:
var duration = this.reproductor.getDuration();
var seconds = this.reproductor.getCurrentTime();
it returns a promise that contains an undefined value even after it is resolved.

The error is under await word, it's like my compiler doesn't understand it, even when I'm using es6 with babel in webpack, and I've defined the async keyword on the function signature declaration.

any thoughts?
